audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,527 @@
+      // It is held artificially low for easy inspection of audio buffers.
+      const sampleRate = 4096;
+      const testDurationSec = 1;
+      const testFrames = testDurationSec * sampleRate;
+      const bufferSize = 512;
+      const scriptProcessorLatency = bufferSize * 2;
+
+      // An appropriate default setting for alpha depends on the frequency
+      // spectrum of the signal and its amplitude relative to the threshold.
+      // When the signal is below the threshold, the gate should be closed. But
+      // with lower frequency signals, the gate will not stay fully closed if it
+      // is very near the threshold. For example, if the oscillator frequency is
+      // set to 40 and alpha is set to 0.90, then the gate won't remain fully
+      // closed at an amplitude greater than 82% of the threshold. 
+      // A lower alpha leads to a lower envelope latency and a smaller
+      // difference between when the gate is expected to be fully closed and
+      // when it actually is fully closed. When frequency is 440Hz and the
+      // signal gain is 90% of the threshold, the alpha setting with shortest 
+      // delay is 0.64 to 0.68 but with an alpha of 0.63 the noise gate will 
+      // not fully close because of high frequency oscillation, implying that 
+      // it is wiser to chose a value higher. When the
+      // frequency of the oscillator is is set to 220, optimal but 'risky' alpha
+      // is 0.82, and when the oscillator oscillates at 110, it is 0.92. Thus
+      // barring a method which adapts alpha depending on a known frequency
+      // range, we set alpha at 0.90, where we expect slight delay but the noise
+      // gate should take action even when lower frequency sounds are near the
+      // threshold.
+      const alpha = 0.90;
+
+      // The threshold for envelope and attack and release tests is some number
+      // between the aboveThresholdGain and belowThresholdGain. If the threshold
+      // is close to the belowThresholdGain, then the gate may flutter between
+      // opened and closed, causing the test to fail.
+      const threshold = 0.2;
+      const aboveThresholdGain = threshold * 1.25;
+      const belowThresholdGain = threshold * 0.75;
+
+      let audit = Audit.createTaskRunner(); 
+
+      // Verify that the noise gate gradually mutes the signal according to the
+      // attack parameter, and that it gradually returns the signal to its
+      // original value according to the release parameter. Precise timing is
+      // not confirmed in this test and is separately investigated in the gain
+      // computer tests.
+      audit.define('attack-and-release', (task, should) => { 
+        let context = new OfflineAudioContext(2, testFrames, sampleRate);
+        
+        // Attack and release are set to be larger than buffer size to ensure
+        // that the test success depends on proper noise gate memory between
+        // blocks.
```

Comment:
Its the number of steps the number of samples it will take from when the threshold is hit to when it is fully open or closed. I figured if we set this value to be some value larger than buffersize then our test hinges on memory between blocks.